queries.sql

```diff
@@ -6,3 +6,38 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg>10.5;
 SELECT * FROM animals WHERE neutered=true;
 SELECT * FROM animals WHERE NOT name='Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- adding "species" in animal table
+BEGIN;
+UPDATE animals SET species='unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species='pokemon' WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '01-01-2022';
+SAVEPOINT savepoint_1;
+UPDATE animals set weight_kg=weight_kg * -1;
+SELECT * FROM animals;
+ROLLBACK;
+UPDATE animals SET weight_kg=weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+SELECT * FROM animals;
+
```

Comment:
- [ ] Great job writing this transaction, however, the requirement is to `rollback` to the savepoint you created and not entirely `rollback` all statements before the `rollback`. This means that the statement below that deletes the Ditto entry in the `animals` table won't be effective:

```sql
DELETE FROM animals WHERE date_of_birth > '01-01-2022';
```
Therefore, some query results will automatically be wrong. For the statement to be effective, you need to `rollback` to the savepoint `savepoint_1`. Kindly implement this change 👍 😎